CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc

```diff
@@ -66,32 +62,30 @@ void CaloTPGTranscoderULUT::loadHCALCompress(HcalLutMetadata const& lutMetadata,
 	HcalTrigTowerDetId id(*i); 
 	if(!theTopology->validHT(id)) continue;
 
-
 	unsigned int index = getOutputLUTId(id); 
 
-	if(index >= size){
-	    size=index+1;
-	    outputLUT_.resize(size);
-	    hcaluncomp_.resize(size);
-	}
-
 	const HcalLutMetadatum *meta = lutMetadata.getValues(id);
 	unsigned int threshold	     = meta->getOutputLutThreshold();
 
 	int ieta=id.ieta();
 	int version=id.version();
 	bool isHBHE = (abs(ieta) < theTrigTowerGeometry.firstHFTower(version)); 
 
-	for (unsigned int i = 0; i < threshold; ++i) outputLUT_[index].push_back(0);
-	for (unsigned int i = threshold; i < OUTPUT_LUT_SIZE; ++i){
-	    LUT value =  isHBHE ? analyticalLUT[i] : 
-			 (version==0?linearRctLUT[i]:linearNctLUT[i]);
-	    outputLUT_[index].push_back(value);
+        unsigned int lutsize = getOutputLUTSize(id);
+
+	for (unsigned int i = 0; i < threshold; ++i) outputLUT_[index][i] = 0;
+	for (unsigned int i = threshold; i < lutsize; ++i){
+           LUT value;
+           if (isHBHE and lutsize == QIE8_OUTPUT_LUT_SIZE) {
```

Comment:
@matz-e - move this set of if statements outside of the look over lutsize

Second - it would be more clear if analytical*LUT variables were renamed to reflect their physical meaning (eg, analyticalLUTQIE8 ?)